python/ray/_private/async_compat.py

```diff
@@ -28,8 +28,30 @@ def try_install_uvloop():
 
 
 def is_async_func(func):
-    """Return True if the function is an async or async generator method."""
-    return inspect.iscoroutinefunction(func) or inspect.isasyncgenfunction(func)
+    """
+    Returns True if the passed function is a coroutine or an async generator.
+
+    note: we prefer this version as Python Coroutines != Cython Coroutines
+    and Cython Coroutines are not detected properly by the inspect module
+
+    this is the optimized version from the links below, using the flag as defined by
+    the inspect module
+    https://github.com/cython/cython/blob/master/tests/run/test_coroutines_pep492.pyx#L889
+    https://github.com/hugapi/hug/blob/develop/hug/introspect.py#L33
+
+    The issues above outline the PR where this was added to ensure async flag support
+    on Cython
+    https://github.com/cython/cython/pull/4902
```

Comment:
I am actually curious, if it is the case, is this also fixable by upgrading the internal cython version?